### ISSUE TITLE
No json object support in type-safe client.

### ISSUE THREAD

**Issue by Jacrispys** [smallrye/smallrye-graphql#1482]
No, it's not implemented, yet, and we'd have to decide about what json classes we'd support.

If you don't need the `data` field of your image class, just remove it! In fact, you should remove all fields that you don't need, not only because of the bloat in the payload, it could also be expensive for the server to fetch the data and/or count against your usage/complexity threshold.

_Originally posted by @t1 in https://github.com/smallrye/smallrye-graphql/discussions/1454#discussioncomment-3211724_

**Comment by Jacrispys** [comment]
Will link PR to solve…

**Comment by t1** [comment]
MicroProfile supports JSON-B, so that has to be supported. Other APIs like Jackson are optional.

**Comment by Jacrispys** [comment]
Alright, I'll get a basic IMPL up tmrw if I have time!

**Comment by Jacrispys** [comment]
@t1 i plan on doing my homework tonight and reading src for where I need to PR, could you point me to the package I will mostly be working in for this?

**Comment by Jacrispys** [comment]
bump @t1 im not sure where to look to find where to add a new type to the type safe client.

**Comment by t1** [comment]
The code you'll need to change is mainly in `client/implementation/src/main/java/io/smallrye/graphql/client/impl/typesafe/json`.

Please remember to also add tests, probably to a new `JsonBehavior` class in `client/tck/src/main/java/tck/graphql/typesafe`.

**Comment by Jacrispys** [comment]
Alright, sounds good thanks!

**Comment by Jacrispys** [comment]
Weird thing... When using the Jakarta `JsonObject` to parse, it gives this NPE
![image](https://user-images.githubusercontent.com/69219325/182000054-cde378d4-df20-48e6-a13e-8edaf0f145fe.png)
![image](https://user-images.githubusercontent.com/69219325/182000073-618fef24-a43e-4790-935a-148da652891d.png)

When navigating to the code in my fork, it points to the type being null:
![image](https://user-images.githubusercontent.com/69219325/182000309-09d59e91-c574-4f41-bbe3-af7e9bd43ee7.png)

Which doesn't make sense, because to my knowledge the `type` field should just be getting `object.class.getType()` or something similar, which cannot be null.

**Comment by t1** [comment]
I'm not sure, but it could be that it recurses into the fields of the `JsonObject` class and some of those fields is generic, i.e. a `ParameterizedType` or so. I guess that's too deep, you'll have to stop it from recursing into `JsonObject` when building the query.

**Comment by t1** [comment]
@Jacrispys: Can we close this issue?

**Comment by Jacrispys** [comment]
@t1 Yes, I mostly abandoned the project that required json objects a while ago due to many reasons. If you would still like to support JSON objects in the future feel free to use this issue as reference as well as the adjacent discussion (https://github.com/smallrye/smallrye-graphql/discussions/1454#discussioncomment-3228075), but I do not have the time nor the justification to implement this feature myself.
